### ISSUE TITLE
Async.Sequential is not sequential

### ISSUE THREAD

**Issue by sbacquet** [dotnet/fsharp#7737]
**Repro steps**

1. Run the following test
```
open Xunit
[<Fact>]
let test () =
    let mutable steps : int list = []
    let add1 x = async { 
        steps <- (x*10) :: steps
        do! Async.Sleep(100)
        steps <- (x*10)+1 :: steps
        return x+1 
    }
    [ add1 1; add1 2; add1 3 ] |> Async.Sequential |> Async.RunSynchronously |> ignore
    steps <- steps |> List.rev
    Assert.True( [ 10; 11; 20; 21; 30; 31 ] = steps)
```
2. Observe it failed

**Expected behavior**

The test succeeds : the resulting `steps` list must be ordered (ascending).

**Actual behavior**

The test fails : the resulting `steps` list is not ordered, most likely because the functions were executed in parallel.

**Known workarounds**

Implement a monadic version of sequence for Async.

**Related information**

* Operating system : Windows 10
* .NET Runtime kind : .NET Core 2.2
* Editing Tools : Visual Studio 2017 (15.9.16)

**Comment by cartermp** [comment]
Dupe of #7564 and fixed in #7596

**Comment by lfr** [comment]
@cartermp Is this supposed to be fixed in Core 3.0 (4.7)? I have good reason to believe I still have this issue.

**Comment by cartermp** [comment]
If you're using an older FSharp.Core binary, then probably yes. But if you're targeting .NET core 3.1 (using latest SDK to account for bug fixes) then it's resolved.
